<launch>
    <!-- Yamabiko -->
    <node pkg="ypspur_ros" type="ypspur_ros" name="ypspur_ros">
        <param name="param_file" value="/home/ruan-x/researches/programs/platform/yp-robot-params/robot-params/speego.param" />
        <param name="port" value="/dev/ttyACM1" />
    </node>
    
    <!-- urg -->
    <node name="hokuyo" pkg="hokuyo_node" type="hokuyo_node" respawn="false" output="screen">
        <param name="serial_port" value="/dev/ttyACM0" />
        <param name="angle_min" value="-1.5707963" />
        <param name="angle_max" value="1.5707963" />
    </node>

    <!-- mycode -->
    <node name="dynamic_openai_node" pkg="openai_test" type="dynamic_openai_node" output="screen" />

    <!-- mycode for mux -->
    <node name="cmdvel_mux" pkg="openai_test" type="cmd_vel_mux" output="screen" />

    <!-- Topic mux -->
    <!-- <node pkg="topic_tools" type="mux" name="mux_cmd_vel" args="ypspur_ros/cmd_vel /cmd_vel_source_avoidance /cmd_vel_source_normal mux:=mux_cmd_vel"/> -->


    <!-- <node pkg="topic_tools" type="mux" name="mux_cmd_vel" args="ypspur_ros/cmd_vel /cmd_vel_source_normal /cmd_vel_source_avoidance mux:=mux_cmd_vel"/> -->

</launch>
